src/shogun/distance/Distance.cpp

```diff
@@ -341,93 +264,64 @@ SGMatrix<T> CDistance::get_distance_matrix()
 	int32_t n=get_num_vec_rhs();
 
 	int64_t total_num = int64_t(m)*n;
+	int64_t total=0;
+	int64_t total_start=0;
+	int64_t total_end=total_num;
 
 	// if lhs == rhs and sizes match assume k(i,j)=k(j,i)
 	bool symmetric= (lhs && lhs==rhs && m==n);
 
 	SG_DEBUG("returning distance matrix of size %dx%d\n", m, n)
 
-		result=SG_MALLOC(T, total_num);
+	result=SG_MALLOC(T, total_num);
 
-	// TODO: port this to use OpenMP
-#ifdef HAVE_PTHREAD
-	int32_t num_threads=parallel->get_num_threads();
-#else
-	int32_t num_threads=1;
-#endif
-	if (num_threads < 2)
-	{
-		D_THREAD_PARAM<T> params;
-		params.distance=this;
-		params.result=result;
-		params.start=0;
-		params.end=m;
-		params.total_start=0;
-		params.total_end=total_num;
-		params.n=n;
-		params.m=m;
-		params.symmetric=symmetric;
-		params.verbose=true;
-		get_distance_matrix_helper<T>((void*) &params);
-	}
-	else
+	int32_t num_threads;
+	int64_t step;
+	#pragma omp parallel shared(num_threads, step)
 	{
-#ifdef HAVE_PTHREAD
-		pthread_t* threads = SG_MALLOC(pthread_t, num_threads-1);
-		D_THREAD_PARAM<T>* params = SG_MALLOC(D_THREAD_PARAM<T>, num_threads);
-		int64_t step= total_num/num_threads;
+		#pragma omp single
+		{
+			num_threads = omp_get_num_threads();
+			step = total_num/num_threads;
+			num_threads--;
+		}
 
-		int32_t t;
+		int32_t thread_num = omp_get_thread_num();
```

Comment:
so isn't better using the actual number of threads spawned and in case of no openmp setting `num_threads=1`?